src/server/verification/verification.js

```diff
@@ -19,49 +20,48 @@ class Verifications implements VerificationAPI {
   /**
    * Verifies user
    * @param {UserRecord} user user details of the user going through FR
-   * @param {*} verificationData data from zoomsdk
+   * @param sessionId
+   * @param imagesBase64
+   * @param storage
 
    */
-  async verifyUser(user: UserRecord, verificationData: any) {
+  async verifyUser(user: UserRecord, sessionId: string, imagesBase64: any, storage: StorageAPI) {
     this.log.debug('Verifying user:', { user })
-    const sessionId = verificationData.sessionId
-    this.log.debug('sessionId:', { sessionId })
-    const searchData = Helper.prepareSearchData(verificationData)
-    // log.info('searchData', { searchData })
-    const isDuplicate = await Helper.isDuplicatesExist(searchData, verificationData.enrollmentIdentifier)
-    GunDBPublic.gun
-      .get(sessionId)
-      .get('isNotDuplicate')
-      .put(!isDuplicate) // publish to subscribers
 
-    this.log.debug('isDuplicate result:', { user: user.identifier, isDuplicate })
-    if (isDuplicate) return { ok: 1, isDuplicate }
-    const enrollData = Helper.prepareEnrollmentData(verificationData)
-    // log.info('enrollData', { enrollData })
-    const enrollResult: EnrollResult = await Helper.enroll(enrollData)
-    GunDBPublic.gun
-      .get(sessionId)
-      .get('isEnrolled')
-      .put(enrollResult.alreadyEnrolled || (enrollResult.enrollmentIdentifier ? true : false)) // publish to subscribers
-    const livenessFailed = (enrollResult && enrollResult.ok === false) || enrollResult.livenessResult === 'undetermined'
-    this.log.debug('liveness result:', { user: user.identifier, livenessFailed })
+    const enrollHandler = async (userId, sessionId, data) => {
+      this.log.debug('Verifying user enrollHandler :', data)
+      const checkFields = ['isDuplicate', 'isLive', 'isEnrolled']
 
-    GunDBPublic.gun
-      .get(sessionId)
-      .get('isLive')
-      .put(!livenessFailed) // publish to subscribers
-    if (livenessFailed) return { ok: 1, livenessPassed: false }
-
-    //this.log.debug('liveness result:', { user: user.identifier, livenessPassed }) // This is left to support future granularity for user better UX experience. Consider using authenticationFacemapIsLowQuality property https://dev.zoomlogin.com/zoomsdk/#/webservice-guide
-    //if (!livenessPassed) return { ok: 1, livenessPassed }
+      for (let field in data) {
+        if (checkFields.indexOf(field) >= 0) {
+          GunDBPublic.gun
+            .get(sessionId)
+            .get(field)
+            .put(data[field])
+        }
+      }
 
-    const isVerified =
-      !isDuplicate && (enrollResult.alreadyEnrolled || (enrollResult.enrollmentIdentifier ? true : false)) // enrollResult.enrollmentIdentifier should return true if there is value in it (and not the value itself) into isVerified.
-    return {
-      ok: 1,
-      isVerified,
-      enrollResult: { ...enrollResult, enrollmentIdentifier: verificationData.enrollmentIdentifier }
+      if (data.ok && data.result) {
```

Comment:
this needs to check that user was enrolled/verified correctly
is testing ok and result follows this logic? is ok true only when user was verified? what does result field contains?
also an "else" is missing for this if

Reply:
1. the same question - what if 'not ok' ? 
- why session is updated to enroll payload **before** checking data.ok && data.result ?
- what shall we do if enroll response fails ? Probably, update session with error message and isWhitelisted: false flag like you do in line 59?

2. GunDBPublic.gun.get(sessionId) ref is used many times so it can be assigned to some constant

Reply:
> 1  the same question - what if 'not ok' ?

I added the condition if data.ok === 0 

> why session is updated to enroll payload before checking data.ok && data.result ?

checking for duplication and for living or not returns {isDuplicate, isLive}
![image](https://user-images.githubusercontent.com/8187382/77072961-2b267280-69f7-11ea-91c0-5b7579f88d40.png)

> GunDBPublic.gun.get(sessionId) ref is used many times so it can be assigned to some constant

**done** use **sessionRef**, done

Reply:
@StanislavShevchenko @serdiukov-o-nordwhale you didnt check the logic like requested
user should be whitelisted only if isEnroll is true.
ok==1 if no exception happened
ok==0 if exception occured